plugins/modules/ec2_spot_instance_info.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/python
+# This file is part of Ansible
+# GNU General Public License v3.0+ (see COPYING or https://wwww.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: ec2_spot_instance_info
+version_added: 2.0.0
+short_description: Gather information about ec2 spot instance requests
+description:
+  - Describes the specified Spot Instance requests.
+author:
+  - Mandar Vijay Kulkarni (@mandar242)
+options:
+  filters:
+    description:
+      - A list of filters to apply.
+      - Each list item is a dict consisting of a filter key and a filter value.
+      - See U(https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeSpotInstanceRequests.html) for possible filters.
+    type: list
+    elements: dict
+  dry_run:
```

Comment:
We don't generally use boto3's dry run features. We should be catching all boto errors to the user already so we're covered for this.
We did use it more historically before we had good Ansible check_mode support so you might see it in a few modules still.